### PR TITLE
Auth with tests

### PR DESCRIPTION
**Pull request by RobertMrowiec**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -2,15 +2,35 @@ const bodyParser = require('body-parser')
...
+dotenv.config()
+
 app.use(cors('*'))
 app.use(bodyParser.json())
 
+app.use((req, res, next) => {
+    if (req.url.startsWith('/api')) {
+        if (req.headers.authorization) {
+            let token = req.headers.authorization.split(' ')[1]
+            jwt.verify(token, process.env.privateKey, (err, decoded) => {
```
Comment (Rxbsxn) [inline app.js]: Maybe we can move this `auth` middleware to separated file and import it here? What do you think?

Comment (RobertMrowiec) [reply app.js]: Yea, sounds great

File: middleware.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+const jwt = require('jsonwebtoken')
+
+module.exports.middlewareLogin = (req, res, next) => {
```
Comment (Rxbsxn) [inline middleware.js]: The method has nothing common with login behavior. Here you are checking that header has `authorization` key and valid it if exists. I'd rather name it `auth` middleware. What do you think?

**Comment by Rxbsxn** [comment]
@RobertMrowiec Could you check out my latest comment?

File: middleware.js (hunk 2 above)
Comment (RobertMrowiec) [reply middleware.js]: sure, that's right it has more with auth than login

Comment (RobertMrowiec) [reply middleware.js]: What Do you think about naming this function just "auth", cause it's already in middleware file

Comment (Rxbsxn) [reply middleware.js]: Totally agree 👍

**Pull request merged**